### ISSUE TITLE
Unreachable code: can't decide kind: Type

### ISSUE THREAD

**Issue by carbotaniuman** [twistedfall/opencv-rust#157]
```
=== Clang: clang version 7.0.1-8 (tags/RELEASE_701/final)
Generated: bioinspired
Generated: bgsegm
Generated: aruco

--- stderr
=== Crate version: Some("0.43.1")
=== Environment configuration:
===   OPENCV_HEADER_DIR = None
===   OPENCV_PACKAGE_NAME = None
===   OPENCV_PKGCONFIG_NAME = None
===   OPENCV_CMAKE_NAME = None
===   OPENCV_CMAKE_BIN = None
===   OPENCV_VCPKG_NAME = None
===   OPENCV_LINK_LIBS = None
===   OPENCV_LINK_PATHS = None
===   OPENCV_INCLUDE_PATHS = None
===   OPENCV_DISABLE_PROBES = None
===   OPENCV_CLANG_STDLIB_PATH = None
===   CMAKE_PREFIX_PATH = None
===   OpenCV_DIR = None
===   PKG_CONFIG_PATH = None
===   VCPKG_ROOT = None
=== Enabled features:
===   BINDING_GENERATOR
===   BUILDTIME_BINDGEN
===   CLANG
===   JOBSERVER
===   OPENCV_34
=== Probing OpenCV library using pkg_config
=== OpenCV library configuration: Library {
    include_paths: [
        "/usr/local/include/opencv",
        "/usr/local/include",
    ],
    version: "3.4.10",
    cargo_metadata: [
        "cargo:rustc-link-search=native=/usr/local/lib",
        "cargo:rustc-link-lib=opencv_stitching",
        "cargo:rustc-link-lib=opencv_superres",
        "cargo:rustc-link-lib=opencv_videostab",
        "cargo:rustc-link-lib=opencv_aruco",
        "cargo:rustc-link-lib=opencv_bgsegm",
        "cargo:rustc-link-lib=opencv_bioinspired",
        "cargo:rustc-link-lib=opencv_ccalib",
        "cargo:rustc-link-lib=opencv_dnn_objdetect",
        "cargo:rustc-link-lib=opencv_dpm",
        "cargo:rustc-link-lib=opencv_highgui",
        "cargo:rustc-link-lib=opencv_videoio",
        "cargo:rustc-link-lib=opencv_face",
        "cargo:rustc-link-lib=opencv_freetype",
        "cargo:rustc-link-lib=opencv_fuzzy",
        "cargo:rustc-link-lib=opencv_hfs",
        "cargo:rustc-link-lib=opencv_img_hash",
        "cargo:rustc-link-lib=opencv_line_descriptor",
        "cargo:rustc-link-lib=opencv_optflow",
        "cargo:rustc-link-lib=opencv_reg",
        "cargo:rustc-link-lib=opencv_rgbd",
        "cargo:rustc-link-lib=opencv_saliency",
        "cargo:rustc-link-lib=opencv_stereo",
        "cargo:rustc-link-lib=opencv_structured_light",
        "cargo:rustc-link-lib=opencv_phase_unwrapping",
        "cargo:rustc-link-lib=opencv_surface_matching",
        "cargo:rustc-link-lib=opencv_tracking",
        "cargo:rustc-link-lib=opencv_datasets",
        "cargo:rustc-link-lib=opencv_text",
        "cargo:rustc-link-lib=opencv_dnn",
        "cargo:rustc-link-lib=opencv_plot",
        "cargo:rustc-link-lib=opencv_xfeatures2d",
        "cargo:rustc-link-lib=opencv_shape",
        "cargo:rustc-link-lib=opencv_video",
        "cargo:rustc-link-lib=opencv_ml",
        "cargo:rustc-link-lib=opencv_ximgproc",
        "cargo:rustc-link-lib=opencv_xobjdetect",
        "cargo:rustc-link-lib=opencv_objdetect",
        "cargo:rustc-link-lib=opencv_calib3d",
        "cargo:rustc-link-lib=opencv_imgcodecs",
        "cargo:rustc-link-lib=opencv_features2d",
        "cargo:rustc-link-lib=opencv_flann",
        "cargo:rustc-link-lib=opencv_xphoto",
        "cargo:rustc-link-lib=opencv_photo",
        "cargo:rustc-link-lib=opencv_imgproc",
        "cargo:rustc-link-lib=opencv_core",
    ],
}
=== Found OpenCV library version: 3.4.10 in headers located at: /usr/local/include
=== Using OpenCV headers from: /usr/local/include/opencv2
=== Generating code in: /home/pi/Desktop/opencv/target/debug/build/opencv-53e71e3be2caf7e6/out
=== Placing generated bindings into: /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.43.1/bindings/rust/opencv_34
=== Clang command line args: [
    "-isystem/usr/bin/../lib/gcc/aarch64-linux-gnu/8/../../../../include/c++/8",
    "-isystem/usr/bin/../lib/gcc/aarch64-linux-gnu/8/../../../../include/aarch64-linux-gnu/c++/8",
    "-isystem/usr/bin/../lib/gcc/aarch64-linux-gnu/8/../../../../include/c++/8/backward",
    "-isystem/usr/include/clang/7.0.1/include",
    "-isystem/usr/local/include",
    "-isystem/usr/include/aarch64-linux-gnu",
    "-isystem/usr/include",
    "-I/usr/local/include",
    "-I/home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.43.1/src_cpp",
    "-DOCVRS_PARSING_HEADERS",
    "-includeocvrs_resolve_types.hpp",
]
thread '<unnamed>' panicked at 'internal error: entered unreachable code: Can't decide kind: Type {
    kind: Half,
    display_name: "__fp16",
}', /home/pi/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-binding-generator-0.16.0/src/type_ref.rs:397:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 2014497936
```

**Comment by twistedfall** [comment]
Are you compiling it directly on Pi? Wait kind of distro are you running? And how did you install OpenCV and LLVM?

**Comment by carbotaniuman** [comment]
I'm running on a Pi 3b+ bit with the experimental 64-bit OS. I built OpenCV (3.4.10) from source, and installed `sudo apt-get install -y llvm-dev libclang-dev clang`.

**Comment by twistedfall** [comment]
Hey, I've pushed a possible fix (just ignoring the __fp16 type) to the `no_half` branch, can you check it out?

**Comment by carbotaniuman** [comment]
It compiles. thanks.

**Comment by twistedfall** [comment]
Fixed in 0.43.2
